### ISSUE TITLE
Optional date

### ISSUE THREAD

**Issue by joostfunkekupper** [theoriginalbit/VariableDateCodable#6]
I'm looking to have an optional `Date` but that won't match the `wrapperValue` type.

```
Property type 'Date?' does not match 'wrappedValue' type 'Date'
```

Any suggestions besides providing a default value?

**Comment by theoriginalbit** [comment]
Unfortunately no, this doesn't support optionals. It basically would require a whole duplication, everything from the DateValue property wrapper to the decoding strategies all assume a non-optional `Date`.

**Comment by joostfunkekupper** [comment]
No worries. Assumed it would probably not easily be possible to do.

**Comment by theoriginalbit** [comment]
I've released a new version (tagged 2.0.0) which supports optionals. You will need to use a new property wrapper however – `@OptionalDateCodable`.

Found a way to have a single strategy definition to work with both optional and non-optional contexts, so any custom strategies can remain the same.
